lucene/core/src/test/org/apache/lucene/analysis/TestCharArraySet.java

```diff
@@ -61,15 +61,17 @@ public void testNonZeroOffset() {
   public void testObjectContains() {
     CharArraySet set = new CharArraySet(10, true);
     Integer val = Integer.valueOf(1);
+    @SuppressWarnings("deprecation")
+    Integer val1 = new Integer(1);
```

Comment:
Add a comment that we're explicitly avoiding the Integer cache, and an `assertNotSame(val, val1)`?